instrumentation/alibaba-druid/src/main/java/brave/druid/TracingStatementFilter.java

```diff
@@ -0,0 +1,230 @@
+package brave.druid;
+
+import brave.Span;
+import brave.propagation.ThreadLocalSpan;
+import com.alibaba.druid.filter.FilterEventAdapter;
+import com.alibaba.druid.proxy.jdbc.ResultSetProxy;
+import com.alibaba.druid.proxy.jdbc.StatementProxy;
+
+import java.net.URI;
+import java.sql.SQLException;
+
+/**
+ * A DB statement interceptor that will report to Zipkin how long each statement takes.
+ */
+public class TracingStatementFilter extends FilterEventAdapter {
+
+
+    private String zipkinServiceName;
+
+    public TracingStatementFilter(String zipkinServiceName){
+        this.zipkinServiceName = zipkinServiceName;
+    }
+
+    public TracingStatementFilter(){
+
+    }
+
+
+    @Override
+    protected void statementExecuteUpdateBefore(StatementProxy statement, String sql) {
+
+        super.statementExecuteUpdateBefore(statement,sql);
+
+        try {
```

Comment:
there are some concerns in this file. I can see you are doing some defensive code, but possibly there are ways to do this in less code. for example, if your before and after blocks did the defense, the delegation methods would not need to.

Reply:
ok